### PR TITLE
Check path when start daemon

### PR DESCRIPTION
**Pull request by vvb2060**

### CODE REVIEW

**Comment by vvb2060** [comment]
https://github.com/topjohnwu/Magisk/blob/0cc29350a0e27ab1081752a55e4d170a235be31c/native/jni/core/restorecon.cpp#L79-L84

7fc9b90

**Comment by canyie** [comment]
https://github.com/topjohnwu/Magisk/commit/4771c2810baf4f3910d9fa0cb94551ab5c21cdef

**Review by gildasle — APPROVED** [review]

File: native/jni/core/daemon.cpp [hunk 1]
```diff
@@ -416,6 +416,14 @@ int connect_daemon(int req, bool create) {
             return -1;
         }
 
+        char buf[64];
+        xreadlink("/proc/self/exe", buf, sizeof(buf));
+        if (strcmp(dirname(buf), "/system/bin") == 0) {
```
Comment (vvb2060) [inline native/jni/core/daemon.cpp:421]:
```suggestion
        if (str_starts(buf, “/system/bin/”)) {
```

File: native/jni/core/daemon.cpp [hunk 2]
```diff
@@ -416,6 +416,14 @@ int connect_daemon(int req, bool create) {
             return -1;
         }
 
+        char buf[64];
+        xreadlink("/proc/self/exe", buf, sizeof(buf));
+        if (str_starts(buf, “/system/bin/”)) {
```
Comment (vvb2060) [inline native/jni/core/daemon.cpp:421]:
```suggestion
        if (str_starts(buf, "/system/bin/")) {
```

**Review by topjohnwu — APPROVED** [review]

**Pull request merged**